### PR TITLE
Use block timeout in events integration test

### PR DESCRIPTION
**Pull request by jomuel**
Replace `gevent.Timeout` by `BlockTimeout` in integration tests where it makes sense, to make sure we wait long enough in every test run.

Related issue: #4524

Edit: The issue has been postponed to the next milestone, so this pr is only for the part of the tests that have been adapted so far. There will be a follow-up pr for the remaining tests.

### CODE REVIEW

**Comment by LefterisJP** [comment]
@hackaugusto please provide some feedback on the approach here

File: raiden/storage/restore.py [hunk 1]
```diff
@@ -210,9 +210,12 @@ def get_event_with_balance_proof_by_locksroot(
 def get_state_change_with_transfer_by_secrethash(
     storage: SerializedSQLiteStorage, secrethash: SecretHash
 ) -> Optional[StateChangeRecord]:
-    filters = [{"transfer.lock.secrethash": to_hex(secrethash)}]
+    filters = [
+        {"from_transfer.lock.secrethash": to_hex(secrethash)},
+        {"transfer.lock.secrethash": to_hex(secrethash)},
+    ]
```
Comment (karlb) [inline raiden/storage/restore.py]: I don't yet understand why we need this change. How can I get an overview of how different state changes look?

Comment (jomuel) [reply raiden/storage/restore.py]: We want to detect both `ActionInitMediator` (which has a `from_transfer` field), and `ActionInitTarget` (which has a `transfer` field) here.

https://github.com/raiden-network/raiden/blob/8fc24e7663a99b4e99917ef21a46df7c51e8732e/raiden/transfer/mediated_transfer/state_change.py#L41

Comment (karlb) [reply raiden/storage/restore.py]: Makes sense. Thanks for the pointer.

File: raiden/tests/unit/test_sqlite.py [hunk 2]
```diff
@@ -362,6 +363,58 @@ def test_get_event_with_balance_proof():
...
+    transfer1 = factories.create(
+        factories.LockedTransferSignedStateProperties(
+            secret=secret1,
+            target=channels.partner_address(1),
+            initiator=channels.partner_address(0),
+        )
+    )
+    state_change1 = factories.mediator_make_init_action(channels, transfer1)
+
+    secret2, secrethash2 = factories.make_secret_with_hash()
```
Comment (karlb) [inline raiden/tests/unit/test_sqlite.py]: The difference between state_change1 and state_change2 take a while to spot. Could you add a tine comment or choose names like `state_change_mediator` `state_change_target`?

Comment (jomuel) [reply raiden/tests/unit/test_sqlite.py]: Yes, I changed the names.

**Comment by jomuel** [comment]
> How long does a `block_timeout_for_transfer_by_secrethash` usually take to timeout in our testing setup?

Not much longer than the timeouts with multiples of `network_wait`. The block time is one second and we will wait for transfer expiration intervals, so 50-60 blocks?

**Pull request merged**